src/veh_interact.cpp

```diff
@@ -3182,6 +3185,7 @@ void veh_interact::complete_vehicle( player &p )
                 p.add_msg_if_player( _( "You completely dismantle the %s." ), veh->name );
                 p.activity.set_to_null();
                 g->m.destroy_vehicle( veh );
+                safe_to_cache = false;
             } else {
                 veh->remove_part( vehicle_part );
                 veh->part_removal_cleanup();
```

Comment:
it would be simpler to move the `update_vehicle_cache()` call to the end of this if block.

Reply:
My concern with that was that it could mess with other things that were relying on that call to update_vehicle_cache() for caching.

Reply:
It won't change anything about how this code works - the caching is only triggered in that else branch, because the precondition to it is set to false in the non-else branch.

Nothing that happens between that else branch and the caching happening would be impacted/impact the caching.

Reply:
Alright, I'll make the change.